lib/arrays.js

```diff
@@ -107,10 +107,26 @@ function anyAreEmpty(...arr) {
   return areEmpty;
 }
 
+/**
+ * Create a deep copy of an array of objects
+ * @param {Array} arr 
+ * @returns {Array} deep copy of arr
+ */
+function deepCopyArrayOfObjects(arr) {
+  let newArr = []
```

Comment:
need param check here to make sure is array of object. you can use `arrTypeCheck` iirc